pkg/hive/health/command.go

```diff
@@ -0,0 +1,147 @@
+		script.CmdUsage{
+			Summary: "List registered metrics",
+			Args:    "[reporter-id-prefix]",
+			Flags: func(fs *pflag.FlagSet) {
+				fs.StringP("match", "m", "", "Output only health reports where the reporter ID path contains the substring")
+				fs.StringArrayP("levels", "s", []string{types.LevelOK, types.LevelDegraded, types.LevelDegraded},
+					"Output only health reports with the specified state (i.e. ok,degraded,stopped)")
+				fs.StringP("file", "f", "", "File to write output to")
+			},
+			Detail: []string{
+				"Prints out a health reporter tree.\n",
+				"If passed prefix is not-empty then only nodes of this subtree\n",
+				"will be displayed\n",
+			},
+		},
+		func(s *script.State, args ...string) (script.WaitFunc, error) {
+			var prefix string
+			if len(args) > 0 {
+				prefix = args[0]
+			}
+
+			match, err := s.Flags.GetString("match")
+			if err != nil {
+				return nil, err
+			}
+
+			levels, err := s.Flags.GetStringArray("levels")
+			if err != nil {
+				return nil, err
+			}
+
+			file, err := s.Flags.GetString("file")
+			if err != nil {
+				return nil, err
+			}
+
+			for i := range levels {
+				levels[i] = strings.ToLower(levels[i])
+			}
+
+			w := s.LogWriter()
+			if file != "" {
+				p := s.Path(file)
+				fd, err := os.Create(p)
+				if err != nil {
+					return nil, err
+				}
+				w = fd
+			}
+
+			ss := getHealth(db, table, prefix, match, levels)
+			healthPkg.GetAndFormatModulesHealth(w, ss, true, "")
+			return nil, nil
+		},
+	)
+}
+
+func getHealth(db *statedb.DB, table statedb.Table[types.Status], prefix, match string, levels []string) []types.Status {
+	ss := []types.Status{}
+	if prefix != "" {
+		tx := db.ReadTxn()
+		for status := range table.Prefix(tx, PrimaryIndex.Query(types.HealthID(prefix))) {
+			ss = append(ss, status)
+		}
+	} else {
+		tx := db.ReadTxn()
+		for status := range table.All(tx) {
+			if match != "" && !strings.Contains(status.ID.String(), match) {
+				continue
+			}
+
+			if !slices.Contains(levels, strings.ToLower(status.Level.String())) {
+				continue
+			}
+
+			ss = append(ss, status)
+		}
+	}
+	return ss
+}
+
+func allOK(db *statedb.DB, table statedb.Table[types.Status]) script.Cmd {
+	return script.Command(
+		script.CmdUsage{
+			Summary: "Report and fail if there are degraded health reports",
+			Args:    "[reporter-id-prefix]",
+			Flags: func(fs *pflag.FlagSet) {
+				fs.StringP("match", "m", "", "Output only health reports where the reporter ID path contains the substring")
+			},
+			Detail: []string{
+				"Checks that all specified health reporters are healthy.\n",
+				"If a non empty prefix is passed, only sub-trees of that .\n",
+				"reporter will be checked\n",
+			},
+		},
+		func(s *script.State, args ...string) (script.WaitFunc, error) {
+			var prefix string
+			if len(args) > 0 {
+				prefix = args[0]
+			}
+
+			match, err := s.Flags.GetString("match")
+			if err != nil {
+				return nil, err
+			}
+
+			w := s.LogWriter()
```

Comment:
We might want to return a `WaitFunc` instead and write to the `stdout` return value. This way we can match on this in tests with the "stdout" command.